mpmp/prediction/survival.py

```diff
@@ -56,27 +65,44 @@ def train_survival(X_train,
         cox.fit(X_train, _y_df_to_struct(y_train))
         alphas = cox.alphas_
 
-    surv_parameters = {
-        "survival__alphas": [[a] for a in alphas],
-        "survival__l1_ratio": l1_ratios,
-    }
-
-    estimator = Pipeline(
-        steps=[
-            (
-                "survival",
-                CoxnetSurvivalAnalysis(
-                    max_iter=max_iter,
-                    tol=1e-5,
-                    # normalize input features
-                    # this seems to help with model convergence
-                    # TODO could try doing this using StandardScaler pipeline
-                    normalize=True,
-                    fit_baseline_model=output_fn
-                ),
-            )
-        ]
-    )
+    if fit_ridge:
+        surv_parameters = {
+            "survival__alpha": alphas
+        }
+        estimator = Pipeline(
+            steps=[
+                (
+                    "survival",
+                    # TODO this isn't currently normalizing features, might want
+                    # to call StandardScaler before this
+                    CoxPHSurvivalAnalysis(
+                        n_iter=max_iter,
+                        tol=1e-5,
+                    ),
+                )
+            ]
+        )
+    else:
+        surv_parameters = {
+            "survival__alphas": [[a] for a in alphas],
+            "survival__l1_ratio": l1_ratios,
+        }
+        estimator = Pipeline(
+            steps=[
+                (
+                    "survival",
+                    CoxnetSurvivalAnalysis(
+                        max_iter=max_iter,
+                        tol=1e-5,
+                        # normalize input features
+                        # this seems to help with model convergence
+                        # TODO could try doing this using StandardScaler pipeline
+                        normalize=True,
```

Comment:
I'm not familiar with Cox survival analysis. Are the results between your two versions here comparable since you normalize one but not the other?

Reply:
Yeah, this is a good point - I should clean this up. Really we should be normalizing all the input features using something like `StandardScaler` (other than binary features which are probably fine not to standardize) and not using the internal `scikit-survival` normalization at all.

I don't expect this to affect the results much, if at all, though.

Reply:
I reran these experiments using `StandardScaler` upstream in my pipeline for all non-binary covariates, and the results were essentially identical. It made the code much cleaner, though!